test/test_geometry.jl

```diff
@@ -243,10 +243,32 @@ import GeoFormatTypes as GFT
     end
 
     @testset "Testing construction of complex geometries" begin
-        @test AG.toWKT(AG.createlinestring([1.0f0, 2.0f0, 3.0f0], [4.0f0, 5.0f0, 6.0f0])) ==
-              AG.toWKT(AG.createlinestring([1.0, 2.0, 3.0], [4.0, 5.0, 6.0])) ==
-              "LINESTRING (1 4,2 5,3 6)"
-        AG.createlinestring([1.0, 2.0, 3.0], [4.0, 5.0, 6.0]) do geom
+
+        @testset "linestring" begin
+            @test AG.toWKT(AG.createlinestring([1.0f0, 2.0f0, 3.0f0], [4.0f0, 5.0f0, 6.0f0])) ==
+                  AG.toWKT(AG.createlinestring([1.0, 2.0, 3.0], [4.0, 5.0, 6.0])) ==
+                  "LINESTRING (1 4,2 5,3 6)"
+            AG.createlinestring([1.0, 2.0, 3.0], [4.0, 5.0, 6.0]) do geom
+                @test GI.geomtrait(geom) == GI.LineStringTrait()
+                @test GI.testgeometry(geom)
+                @test isapprox(
+                    GI.coordinates(geom),
+                    [[1, 4], [2, 5], [3, 6]],
+                    atol = 1e-6,
+                )
+                @test AG.toWKT(geom) == "LINESTRING (1 4,2 5,3 6)"
+                AG.closerings!(geom)
+                @test AG.toWKT(geom) == "LINESTRING (1 4,2 5,3 6)"
+                AG.setpoint!(geom, 1, 10, 10)
+                @test AG.toWKT(geom) == "LINESTRING (1 4,10 10,3 6)"
+                @test GFT.val(convert(GFT.WellKnownText, geom)) == AG.toWKT(geom)
+                @test typeof(geom) == AG.Geometry{AG.wkbLineString}
+                # child = AG.unsafe_getgeom(geom, 0)
```

Comment:
Why are these lines commented out?

Reply:
Because `unsafe_geom` doesn't work, and I don't know why...

Reply:
> Because `unsafe_geom` doesn't work, and I don't know why...

I have a feeling (i) `getgeom` only works for polygons, geometrycollections, and multi* etc, (ii) getpoint() will work here. But I haven't tested to be sure.

If (ii) is right, maybe we should make getgeom dispatch to getpoint for linestrings?

Reply:
This is correct, `ogr_g_getgeometryref` doesn't work on linestrings.

Reply:
Yeah I think that why we need the code that was deleted in https://github.com/yeesian/ArchGDAL.jl/pull/316/commits/792fd22d38259c9e3be98b44a54a1bbcc7ad75f7.